airflow-core/src/airflow/models/taskinstance.py

```diff
@@ -326,6 +327,19 @@ def clear_task_instances(
             if dr.state in State.finished_dr_states:
                 dr.state = dag_run_state
                 dr.start_date = timezone.utcnow()
+                if TYPE_CHECKING:
+                    assert dag  # todo: change signature so this is required
+                if not dag.disable_bundle_versioning:
+                    if dr.dag_model:
+                        bundle_version = dr.dag_model.bundle_version
+                    else:
+                        bundle_version = session.scalar(
+                            select(DagModel.bundle_version).where(
+                                DagModel.dag_id == dag.dag_id,
+                            )
+                        )
```

Comment:
When is this useful? Some kind of db state handling issue?

Reply:
removed this logic